### DISCUSSION TITLE
use sublime build-in botton in insert mode

### DISCUSSION THREAD

**Discussion by xianyuIC** [NeoVintageous/NeoVintageous#997] [Q&A] [answered]
Can I use sublime's keystrokes when I'm in insert mode? For example, when in insert mode, pressing "Ctrl + F" opens sublime's search screen. For example, you can press Ctrl + c to copy, or Ctrl + v to paste.

That way, when I want to use the vim command, I can press esc to go to visual mode, and I can use the vim button. When I'm in insert mode, I can use some of sublime's built-in buttons that are very handy.

In this way, we can enjoy the benefits of vim in sublime, but also keep the benefits of sublime itself. That would be a great improvement.

**Comment by gerardroche** [comment]
Most insert commands are not implemented because they cause too many conflicts with sublime's normal functionality e.g. overlays, prompts, inputs, etc.

For the ones that are implemented you can use the `:ino[remap] {lhs} {rhs}` mapping: https://github.com/NeoVintageous/NeoVintageous/blob/master/README.md#map-commands.

If you want to make a key mappable in insert mode, add custom keymaps for them, for example consider one defined by NeoVintageous for `ctrl+[`:

```
    {"keys": ["ctrl+["], "command": "nv_feed_key", "args": {"key": "<C-[>"}, "context": [{"key": "nv_command_or_insert"},  {"key": "setting.vintageous_use_ctrl_keys"}, {"key": "nv_handle_key", "operand": "<C-[>"}]},
```

Just replace `ctrl+[` and `<C-[>` with your key.

Here is an experimental package, not available in Package Control, that tries to define all the insert mode keys:

https://github.com/gerardroche/NeoVintageousInsertModeKeymaps

Once you define your keymaps, you can use the insert mapping command `:ino[remap] {lhs} {rhs}`.

**Comment by gerardroche** [comment, marked as answer]
Oh wait, I think I misunderstood you. You want to **disable** NeoVintagesous keys in insert mode? You can do that with the `vintageous_handle_keys` setting:

```

    "vintageous_handle_keys": {
        // "i_<Left>": false,
        // "i_<Right>": false,
        "<C-f12>": false,
        "<S-f1>": false,
        "<S-tab>": false,
        "<f1>": false,
        "i_<C-d>": false,
        "i_<C-u>": false,
    },
```

See https://github.com/NeoVintageous/NeoVintageous#settings and https://blog.gerardroche.com/2022/09/22/neovintageous-key-handler/

**Reply by xianyuIC** [reply]
Thank you for your answer, this solves my problem, thank you !
